src/tests/test_custom_commands.rs

```diff
@@ -145,3 +145,46 @@ fn override_table_eval_file() {
     let actual = nu!(cwd: ".", r#"def table [] { "hi" }; table"#);
     assert_eq!(actual.out, "hi");
 }
+
+// These tests are disabled on Windows because they cause a stack overflow in CI (but not locally!).
+// For reasons we don't understand, the Windows CI runners are prone to stack overflow.
+// TODO: investigate so we can enable these on Windows
+#[cfg(not(target_os = "windows"))]
+#[test]
+fn recursion_successful() {
+    let actual = nu!(
+        cwd: ".",
+        r#"
+            def bang [] { bang };bang
+        "#
+    );
+    assert!(actual.err.is_empty());
+}
+
+#[cfg(not(target_os = "windows"))]
+#[test]
+fn call_def_in_def_should_be_successful() {
+    let actual = nu!(
+        cwd: ".",
+        r#"
+            def bang [b] { let t = bang $b; return $b;echo};bang passing_data
+        "#
+    );
+
+    assert!(actual.out.contains("passing_data"));
+    assert!(actual.err.is_empty());
+}
+
+#[cfg(not(target_os = "windows"))]
+#[test]
+fn call_def_in_def_should_be_successful2() {
+    let actual = nu!(
+        cwd: ".",
+        r#"
+            def slice [a,b] { let t = slice 1 2;return result_is_not_null };slice 1 2 
+        "#
+    );
+
+    assert!(actual.out.contains("result_is_not_null"));
+    assert!(actual.err.is_empty());
```

Comment:
Not sure what this test adds to the one above? A worthy test would be to test if a function that is terminating the recursion properly within the recursion limit passes.